packages/common/src/core/types/DynamicValues.ts

```diff
@@ -0,0 +1,176 @@
+import { IQuery } from "../../search/types/IHubCatalog";
+import { IConfigurationSchema } from "../schemas";
+
+/**
+ * The source for a dynamic value. Used to determine how to resolve the value
+ */
+export type DynamicValueSource =
```

Comment:
should this be `DynamicValueType`, or should we change `IBaseValueDefinition` to have `source: DynamicValueSourece`?